rustls/src/server/hs.rs

```diff
@@ -564,9 +564,9 @@ impl State<ServerConnectionData> for ExpectClientHello {
 ///
 /// [`ResolvesServerCert`]: crate::server::ResolvesServerCert
 pub(super) fn process_client_hello<'a>(
-    m: &'a Message,
+    m: &'a Message<'_>,
```

Comment:
I find this one confusing. In fact I don't think I understand what the lifetimes here actually are or which are different names for the same span. Like we have `Message<'m>` in the immediate caller, but then that is inferred in this function, where `'a` is the caller's scope. Then the return from the caller is `'static`, not `'m`? So can we delete/infer `'m` in the caller?

Reply:
I think it would make sense to use `'a` here (or rename `'a` to '`m`).

Reply:
I think we can infer the `'m` on the `Message<'_>` lifetime in the callsite, but we still need the explicit `'m` there for the `Self` clause used for the `NextStateOrError` return  - it's used to signal the borrow comes from `m` and not `cx` here AFAICT

Something like:

```diff
--- a/rustls/src/server/hs.rs
+++ b/rustls/src/server/hs.rs
@@ -540,7 +540,7 @@ impl State<ServerConnectionData> for ExpectClientHello {
     fn handle<'m>(
         self: Box<Self>,
         cx: &mut ServerContext<'_>,
-        m: Message<'m>,
+        m: Message<'_>,
     ) -> NextStateOrError<'m>
     where
         Self: 'm,
```

Reply:
> (or rename 'a to 'm).

That sounds good to me. I'll push an adjustment.

Reply:
> I'll push an adjustment.

Done